src/util/png_image.hpp

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "files.hpp"
+
+#include <png.h>
+
+#include <filesystem>
+
+namespace image {
+
+template <class T>
+void
+write_image(T image, std::filesystem::path path) {
+    //if (!std::filesystem::exists(path)) [[unlikely]] {
+    //    return;
+    //}
+
+    std::FILE* file = fopen(path.string().data(), "wb");
```

Comment:
```suggestion
    auto path_str = path.string();
    FILE* file = fopen(path_str.c_str(), "wb");
    
    if (!file) [[unlikely]] {
        log_e(images, "Could not open file for reading");
        
        success = false;
        goto fopen_failed;
    }
```

so fun fact

this string gets freed because you didn't save it in a variable

so that was UB